src/components/layout/MainLayout/MainLayout.js

```diff
@@ -1,21 +1,51 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
+import { windowResize } from '../../../redux/deviceRedux';
 import Header from '../Header/Header';
 import Footer from '../Footer/Footer';
 // import Feedback from '../../features/Feedback/Feedback';
 import Feedback from '../../features/Feedback/FeedbackContainer';
 import CompareContainer from '../../features/CompareContainer/CompareContainer';
+import { useDispatch, useSelector } from 'react-redux';
+import { useState, useEffect } from 'react';
+
+const MainLayout = ({ children }) => {
+
+
+  const [width, setWidth] = useState(window.innerWidth);
+  useEffect(() => {
+    const handleWindowResize = () => setWidth(window.innerWidth);
+    window.addEventListener('resize', handleWindowResize);
+    sizeChange();
```

Comment:
Po zmainie szerokości powinno się to dispatchować, a nie tylko przy uruchomieniu aplikacji